astropy/timeseries/downsample.py

```diff
@@ -0,0 +1,136 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import warnings
+
+import numpy as np
+from astropy import units as u
+from astropy.utils.exceptions import AstropyUserWarning
+
+from astropy.timeseries.sampled import TimeSeries
+from astropy.timeseries.binned import BinnedTimeSeries
+
+__all__ = ['simple_downsample']
+
+
+def reduceat(array, indices, function):
+    """
+    Manual reduceat functionality for cases where Numpy functions don't have a reduceat.
+    It will check if the input function has a reduceat and call that if it does.
+    """
+    if hasattr(function, 'reduceat'):
+        return np.array(function.reduceat(array, indices))
+    else:
+        result = []
+        for i in range(len(indices) - 1):
+            if indices[i+1] <= indices[i]+1:
+                result.append(function(array[indices[i]]))
+            else:
+                result.append(function(array[indices[i]:indices[i+1]]))
+        result.append(function(array[indices[-1]:]))
+        return np.array(result)
+
+
+def simple_downsample(time_series, time_bin_size, func=None, time_bin_start=None, n_bins=None):
+    """
+    Downsample a time series by binning values into bins with a fixed size,
+    using a single function
+
+    Parameters
+    ----------
+    time_series : :class:`~astropy.timeseries.TimeSeries`
+        The time series to downsample.
+    time_bin_size : `~astropy.units.Quantity`
+        The time interval for the binned time series
+    func : callable, optional
```

Comment:
should instead be `combine_func` or possibly `bin_combine_func` or similar.  Just `func` doesn't give enough context about what the function is for.

Reply:
Fixed